src/server/routes/parcel.routes.ts

```diff
@@ -0,0 +1,66 @@
+import type { FastifyInstance, RouteOptions } from 'fastify';
+import { Parcel } from '@relaycorp/relaynet-core';
+
+import type { PluginDone } from '../../utilities/fastify/PluginDone.js';
+import { bufferToArrayBuffer } from '../../utilities/buffer.js';
+import { HTTP_STATUS_CODES } from '../../utilities/http.js';
+
+export default function registerRoutes(
+  fastify: FastifyInstance,
+  _opts: RouteOptions,
+  done: PluginDone,
+): void {
+  fastify.removeAllContentTypeParsers();
+  fastify.addContentTypeParser(
+    'application/vnd.awala.parcel',
+    { parseAs: 'buffer' },
+    (_request, payload, next) => {
+      next(null, payload);
+    },
+  );
+
+  interface Request {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    readonly Body: Buffer;
+  }
+
+  fastify.route<Request>({
+    method: ['POST'],
+    url: '/',
+
+    async handler(request, reply): Promise<void> {
+      let parcel;
+      try {
+        parcel = await Parcel.deserialize(bufferToArrayBuffer(request.body));
+      } catch (err) {
+        // Don't log the full error because 99.99% of the time the reason will suffice.
+        request.log.info({ reason: (err as Error).message }, 'Refusing malformed parcel');
+        return reply
+          .code(HTTP_STATUS_CODES.FORBIDDEN)
```

Comment:
I just noticed a bug in the original implementation in Pong: A malformed parcel should actually return a `400` response. Please fix it here.